File: generate_icons.py
from PIL import Image
import os

# Icon sizes for different densities
sizes = {
    'mipmap-mdpi': 48,
    'mipmap-hdpi': 72,
    'mipmap-xhdpi': 96,
    'mipmap-xxhdpi': 144,
    'mipmap-xxxhdpi': 192
}

# Paths
source_icon = r'C:\Users\rahul\.gemini\antigravity\brain\10a4bb69-02d1-4011-be53-487c7cb7ad0d\app_icon_512_1768976729936.png'
android_res = r'c:\test antigravity\may-i-borrow\android\app\src\main\res'

# Open the source image
img = Image.open(source_icon)

# Create icons for each density
for folder, size in sizes.items():
    # Create folder if it doesn't exist
    folder_path = os.path.join(android_res, folder)
    os.makedirs(folder_path, exist_ok=True)
    
    # Resize image
    resized = img.resize((size, size), Image.Resampling.LANCZOS)
    
    # Save as ic_launcher.png
    output_path = os.path.join(folder_path, 'ic_launcher.png')
    resized.save(output_path, 'PNG')
    print(f'Created {folder}/ic_launcher.png ({size}x{size})')
    
    # Also save as ic_launcher_foreground.png for adaptive icons
    output_path_fg = os.path.join(folder_path, 'ic_launcher_foreground.png')
    resized.save(output_path_fg, 'PNG')
    print(f'Created {folder}/ic_launcher_foreground.png ({size}x{size})')
    
    # Save as ic_launcher_round.png
    output_path_round = os.path.join(folder_path, 'ic_launcher_round.png')
    resized.save(output_path_round, 'PNG')
    print(f'Created {folder}/ic_launcher_round.png ({size}x{size})')

print('\nAll icons created successfully!')

File: generate_icons.js
const fs = require('fs');
const path = require('path');

// For this script, we'll use PowerShell to resize images
const { execSync } = require('child_process');

const sizes = {
    'mipmap-mdpi': 48,
    'mipmap-hdpi': 72,
    'mipmap-xhdpi': 96,
    'mipmap-xxhdpi': 144,
    'mipmap-xxxhdpi': 192
};

const sourceIcon = String.raw`C:\Users\rahul\.gemini\antigravity\brain\10a4bb69-02d1-4011-be53-487c7cb7ad0d\app_icon_512_1768976729936.png`;
const androidRes = String.raw`c:\test antigravity\may-i-borrow\android\app\src\main\res`;

// PowerShell script to resize image
function resizeImage(source, output, width, height) {
    const psScript = `
        Add-Type -AssemblyName System.Drawing
        $img = [System.Drawing.Image]::FromFile('${source}')
        $bitmap = New-Object System.Drawing.Bitmap($img, ${width}, ${height})
        $bitmap.Save('${output}', [System.Drawing.Imaging.ImageFormat]::Png)
        $bitmap.Dispose()
        $img.Dispose()
    `;

    try {
        execSync(`powershell -Command "${psScript.replace(/\n/g, ' ')}"`, { stdio: 'inherit' });
        console.log(`Created ${output}`);
    } catch (error) {
        console.error(`Failed to create ${output}:`, error.message);
    }
}

// Create icons for each density
for (const [folder, size] of Object.entries(sizes)) {
    const folderPath = path.join(androidRes, folder);

    // Create folder if it doesn't exist
    if (!fs.existsSync(folderPath)) {
        fs.mkdirSync(folderPath, { recursive: true });
    }

    // Create ic_launcher.png
    resizeImage(sourceIcon, path.join(folderPath, 'ic_launcher.png'), size, size);

    // Create ic_launcher_foreground.png
    resizeImage(sourceIcon, path.join(folderPath, 'ic_launcher_foreground.png'), size, size);

    // Create ic_launcher_round.png
    resizeImage(sourceIcon, path.join(folderPath, 'ic_launcher_round.png'), size, size);
}

console.log('\nAll icons created successfully!');
